coalib/__init__.py

```diff
@@ -6,10 +6,69 @@
 
 
 import sys
+from os import chdir, getcwd
 from os.path import join, dirname
+from subprocess import check_output, CalledProcessError
 
 
-VERSION_FILE = join(dirname(__file__), 'VERSION')
+COALIB_DIR = dirname(__file__)
+
+VERSION_FILE = join(COALIB_DIR, 'VERSION')
+
+COMMIT_FILE = join(COALIB_DIR, 'COMMIT')
+
+
+def get_current_commit(git_repo_dir):
+    """
+    :param git_repo_dir:
+        String containing the directory to check the current commit in.
+    :return:
+        String containing the current commit sha if `git_repo_dir` is
+        initialized as a git repo otherwise an empty string.
+    """
+    old_dir = getcwd()
+    cwd_changed = False
+
+    if old_dir != git_repo_dir:
+        chdir(git_repo_dir)
+        cwd_changed = True
+
+    try:
+        commit = check_output(['git', 'rev-parse', 'HEAD'])
+        commit = commit.decode('utf-8').strip()
+
+        status = check_output(['git', 'status', '--porcelain'])
+        status = status.decode('utf-8').strip()
+        if status:
+            commit += '*'
+    except (OSError, CalledProcessError) as e:
+        commit = ''
+
+    if cwd_changed:
+        chdir(old_dir)
+    return commit
+
+
+def get_commit(git_repo_dir, commit_file):
+    """
+    :param git_repo_dir:
+        String containing the directory to check the current commit in.
+    :param commit_file:
+        Filename containing commit sha of HEAD at the time of installation.
+    :return:
+        String containing the current commit sha if `git_repo_dir` is a
+        git repo otherwise commit sha in commit_file.
+    """
+    try:
+        with open(commit_file) as cf:
+            commit = cf.readline().strip()
+    # Not using FileNotFoundError, as python2 CI starts complaining
+    except (OSError, IOError):
+        commit = ''
```

Comment:
`commit = None`.  Do not use the empty string unless it is actually an empty string that you need.